### ISSUE TITLE
[tune] DistributedTrainableCreator can not get access to current trial_id

### ISSUE THREAD

**Issue by pablete** [ray-project/ray#15288]
```
from ray.tune.integration.tensorflow import DistributedTrainableCreator

tf_trainable = DistributedTrainableCreator(
        train_function,
        num_workers=2,
        num_cpus_per_worker=12,
        num_gpus_per_worker=1
)

analysis = tune.run(
        tf_trainable,
        ...
        num_samples=10
)
```
Now inside the train_function

```
def train_function
....
current_session = ray.tune.session.get_session()
logger.info("current_session      => {}".format( current_session ))
logger.info("trial_id             => {}".format( current_session._trial_id ))
logger.info("trial_name           => {}".format( current_session._trial_name ))
.....
```

Output

```
current_session  => <ray.tune.function_runner.StatusReporter object at 0x7ef3712df210>
trial_id         => default 
trial_name       => default
```
In any of the DistributedTrainableCreator (torch/tensorflow/horovod) is the current trial (trial_id) information available.

"default"  in Trainable  happens when you do not receive __trial_info__ in the config

https://github.com/ray-project/ray/blob/master/python/ray/tune/trainable.py#L74

Is there another way to query or pass the Trial information to the actual function that will be executed as a tune.Trainable ?

Thanks

**Comment by richardliaw** [comment]
Ah, thanks for reporting this, @pablete ! this seems like a bug. 

What are you trying to do specifically?

**Comment by pablete** [comment]
I am training a model that requires multi-gpu. The metrics are being calculated outside tensorflow on a metrics_supervisor/metrics_worker set of actors.  That set of actors keep the validation set loaded and receive model / model weights to compute the metrics.  

Since this is computed using a Keras Callback, each tf worker on a multi-gpu setting will trigger the same call (ie tfw1, tfw2, tfw3, tfw4) will request to compute the same metrics for the same model at every epoch.

If I get the unique identifier of the trial_id, I can compute the metrics ONCE by submitting `supervisor.compute_metrics.remote(model_weights, trial_id, epoch)`  and then have all the tensorflow workers pool by supervisor.read_metrics(trial_id, epoch) until the results are there.

My temporary solution is to have `unique_id = hashlib.md5("{}".format(config).encode()).hexdigest()`  so at least each worker (w1, w2, w3, w4) that belong to the SAME trial (trial_id) will have the same config, hence the same unique_id.

Let me know how I can help.

**Comment by richardliaw** [comment]
hey @pablete I'll put this item on my backlog. Good to hear that you have a temp solution!

May I ask what you're using Ray to do?
